File: sdk/nodejs/src/resources/virtualizor.js
"use strict";

const { BaseResource } = require("./base");

class VirtualizorResource extends BaseResource {
  async suspend(serviceId, vmId) {
    return this._http.post("/api/service/{id}/vms/{vmid}/suspend");
  }

  async unsuspend(serviceId, vmId) {
    return this._http.post("/api/service/{id}/vms/{vmid}/unsuspend");
  }

  async getRebuildInfo(serviceId, vmId) {
    return this._http.get("/api/service/{id}/vms/{vmid}/rebuild");
  }

  async addSshKey(serviceId, vmId, data) {
    return this._http.post("/api/service/{id}/vms/{vmid}/addsshkey", data);
  }

}

module.exports = { VirtualizorResource };

File: sdk/csharp/Resources/Virtualizor.cs
namespace HiTechCloud.SDK.Resources;

public class Virtualizor : BaseResource
{
    public Virtualizor(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> SuspendAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/suspend", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> UnsuspendAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/unsuspend", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> RebuildAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/rebuild", id, vmid));
    }

    public async Task<Dictionary<string, object?>> AddSshKeyAsync(int id, int vmid, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/addsshkey", id, vmid), data);
    }
}
